riscv-executor/src/memory.rs

```diff
@@ -51,16 +51,24 @@ impl<F: FieldElement> Submachine<F> for MemoryMachine<F> {
         &self.namespace
     }
 
-    fn add_operation(&mut self, selector_idx: Option<u8>, lookup_args: &[F], _extra: &[F]) {
+    fn add_operation(&mut self, selector: Option<&str>, lookup_args: &[F], _extra: &[F]) {
         let [op_id, addr, step, value] = lookup_args[..] else {
             panic!()
         };
+        // get the idx from the selector
+        let selector_idx = selector
+            .map(|s| {
+                let start = s.find('[').unwrap() + 1;
+                let end = s.find(']').unwrap();
+                s[start..end].parse::<u8>().unwrap()
+            })
+            .unwrap();
```

Comment:
This seems kinda costly if it has to run every time? Wouldn't it be better to have a map in the caller and just pass the number like before?

Reply:
yes, this is ugly, i'm working on a separate PR for these details/performance (this here should be cached)